### ISSUE TITLE
Weird error when trying to use Javacord

### ISSUE THREAD

**Issue by Hexab** [Javacord/Javacord#678]
Hello there,

I am trying to make a discord bot and I get a huge error

Main class:
```java


        DiscordApi api = new DiscordApiBuilder()
                // An inline listener
                .addServerBecomesAvailableListener(event -> {
                    System.out.println("Loaded " + event.getServer().getName());
                })
                .addListener(new BotListeners())
                .setToken("")
                .setWaitForServersOnStartup(false)
                .login()
                .join();
    }
```

Error:
```
2020-12-22 19:25:44.490+0000 INFO org.javacord.core.util.logging.ExceptionLoggerDelegateImpl No Log4j2 compatible logger was found. Using default Javacord implementation!
2020-12-22 19:25:46.970+0000 WARN org.javacord.core.util.gateway.DiscordWebSocketAdapter An error occurred while connecting to websocket {shard=0} 
 java.util.concurrent.CompletionException: java.lang.NoSuchMethodError: okio/ByteString.startsWith(Lokio/ByteString;)Z (loaded from file:/C:/Users/zulif/.m2/repository/com/squareup/okio/okio/1.6.0/okio-1.6.0.jar by sun.misc.Launcher$AppClassLoader@d463c91f) called from class okhttp3.internal.http2.Hpack$Writer (loaded from file:/C:/Users/zulif/.m2/repository/com/squareup/okhttp3/okhttp/3.9.1/okhttp-3.9.1.jar by sun.misc.Launcher$AppClassLoader@d463c91f).
	at java.util.concurrent.CompletableFuture.reportJoin(CompletableFuture.java:375)
	at java.util.concurrent.CompletableFuture.join(CompletableFuture.java:1947)
	at org.javacord.core.util.gateway.DiscordWebSocketAdapter.getGateway(DiscordWebSocketAdapter.java:322)
	at org.javacord.core.util.gateway.DiscordWebSocketAdapter.connect(DiscordWebSocketAdapter.java:367)
	at org.javacord.core.util.gateway.DiscordWebSocketAdapter.<init>(DiscordWebSocketAdapter.java:185)
	at org.javacord.core.util.gateway.DiscordWebSocketAdapter.<init>(DiscordWebSocketAdapter.java:166)
	at org.javacord.core.DiscordApiImpl.lambda$new$7(DiscordApiImpl.java:603)
	at org.javacord.core.DiscordApiImpl$$Lambda$29/000000000000000000.run(Unknown Source)
	at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:511)
	at java.util.concurrent.FutureTask.run(FutureTask.java:266)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624)
	at java.lang.Thread.run(Thread.java:823)
Caused by: java.lang.NoSuchMethodError: okio/ByteString.startsWith(Lokio/ByteString;)Z (loaded from file:/C:/Users/zulif/.m2/repository/com/squareup/okio/okio/1.6.0/okio-1.6.0.jar by sun.misc.Launcher$AppClassLoader@d463c91f) called from class okhttp3.internal.http2.Hpack$Writer (loaded from file:/C:/Users/zulif/.m2/repository/com/squareup/okhttp3/okhttp/3.9.1/okhttp-3.9.1.jar by sun.misc.Launcher$AppClassLoader@d463c91f).
	at okhttp3.internal.http2.Hpack$Writer.writeHeaders(Hpack.java:511)
	at okhttp3.internal.http2.Http2Writer.headers(Http2Writer.java:299)
	at okhttp3.internal.http2.Http2Writer.synStream(Http2Writer.java:127)
	at okhttp3.internal.http2.Http2Connection.newStream(Http2Connection.java:230)
	at okhttp3.internal.http2.Http2Connection.newStream(Http2Connection.java:205)
	at okhttp3.internal.http2.Http2Codec.writeRequestHeaders(Http2Codec.java:111)
	at okhttp3.internal.http.CallServerInterceptor.intercept(CallServerInterceptor.java:50)
	at okhttp3.internal.http.RealInterceptorChain.proceed(RealInterceptorChain.java:147)
	at okhttp3.internal.connection.ConnectInterceptor.intercept(ConnectInterceptor.java:45)
	at okhttp3.internal.http.RealInterceptorChain.proceed(RealInterceptorChain.java:147)
	at okhttp3.internal.http.RealInterceptorChain.proceed(RealInterceptorChain.java:121)
	at okhttp3.internal.cache.CacheInterceptor.intercept(CacheInterceptor.java:93)
	at okhttp3.internal.http.RealInterceptorChain.proceed(RealInterceptorChain.java:147)
	at okhttp3.internal.http.RealInterceptorChain.proceed(RealInterceptorChain.java:121)
	at okhttp3.internal.http.BridgeInterceptor.intercept(BridgeInterceptor.java:93)
	at okhttp3.internal.http.RealInterceptorChain.proceed(RealInterceptorChain.java:147)
	at okhttp3.internal.http.RetryAndFollowUpInterceptor.intercept(RetryAndFollowUpInterceptor.java:125)
	at okhttp3.internal.http.RealInterceptorChain.proceed(RealInterceptorChain.java:147)
	at okhttp3.internal.http.RealInterceptorChain.proceed(RealInterceptorChain.java:121)
	at okhttp3.logging.HttpLoggingInterceptor.intercept(HttpLoggingInterceptor.java:212)
	at okhttp3.internal.http.RealInterceptorChain.proceed(RealInterceptorChain.java:147)
	at okhttp3.internal.http.RealInterceptorChain.proceed(RealInterceptorChain.java:121)
	at org.javacord.core.DiscordApiImpl.lambda$new$1(DiscordApiImpl.java:576)
	at org.javacord.core.DiscordApiImpl$$Lambda$25/000000000000000000.intercept(Unknown Source)
	at okhttp3.internal.http.RealInterceptorChain.proceed(RealInterceptorChain.java:147)
	at okhttp3.internal.http.RealInterceptorChain.proceed(RealInterceptorChain.java:121)
	at okhttp3.RealCall.getResponseWithInterceptorChain(RealCall.java:200)
	at okhttp3.RealCall.execute(RealCall.java:77)
	at org.javacord.core.util.rest.RestRequest.executeBlocking(RestRequest.java:355)
	at org.javacord.core.util.ratelimit.RatelimitManager.lambda$queueRequest$2(RatelimitManager.java:114)
	at org.javacord.core.util.ratelimit.RatelimitManager$$Lambda$42/000000000000000000.run(Unknown Source)
	... 5 more
2020-12-22 19:25:46.983+0000 INFO org.javacord.core.util.gateway.DiscordWebSocketAdapter Trying to reconnect/resume in 1 seconds! {shard=0} 
2020-12-22 19:25:47.986+0000 WARN org.javacord.core.util.gateway.DiscordWebSocketAdapter An error occurred while connecting to websocket {shard=0} 
 java.util.concurrent.CompletionException: java.lang.NoSuchMethodError: okio/ByteString.startsWith(Lokio/ByteString;)Z (loaded from file:/C:/Users/zulif/.m2/repository/com/squareup/okio/okio/1.6.0/okio-1.6.0.jar by sun.misc.Launcher$AppClassLoader@d463c91f) called from class okhttp3.internal.http2.Hpack$Writer (loaded from file:/C:/Users/zulif/.m2/repository/com/squareup/okhttp3/okhttp/3.9.1/okhttp-3.9.1.jar by sun.misc.Launcher$AppClassLoader@d463c91f).
	at java.util.concurrent.CompletableFuture.reportJoin(CompletableFuture.java:375)
	at java.util.concurrent.CompletableFuture.join(CompletableFuture.java:1947)
	at org.javacord.core.util.gateway.DiscordWebSocketAdapter.getGateway(DiscordWebSocketAdapter.java:322)
	at org.javacord.core.util.gateway.DiscordWebSocketAdapter.connect(DiscordWebSocketAdapter.java:367)
	at org.javacord.core.util.gateway.DiscordWebSocketAdapter.lambda$connect$14(DiscordWebSocketAdapter.java:470)
	at org.javacord.core.util.gateway.DiscordWebSocketAdapter$$Lambda$76/000000000000000000.run(Unknown Source)
	at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:511)
	at java.util.concurrent.FutureTask.run(FutureTask.java:266)
	at java.util.concurrent.ScheduledThreadPoolExecutor$ScheduledFutureTask.access$201(ScheduledThreadPoolExecutor.java:180)
	at java.util.concurrent.ScheduledThreadPoolExecutor$ScheduledFutureTask.run(ScheduledThreadPoolExecutor.java:293)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624)
	at java.lang.Thread.run(Thread.java:823)
Caused by: java.lang.NoSuchMethodError: okio/ByteString.startsWith(Lokio/ByteString;)Z (loaded from file:/C:/Users/zulif/.m2/repository/com/squareup/okio/okio/1.6.0/okio-1.6.0.jar by sun.misc.Launcher$AppClassLoader@d463c91f) called from class okhttp3.internal.http2.Hpack$Writer (loaded from file:/C:/Users/zulif/.m2/repository/com/squareup/okhttp3/okhttp/3.9.1/okhttp-3.9.1.jar by sun.misc.Launcher$AppClassLoader@d463c91f).
	at okhttp3.internal.http2.Hpack$Writer.writeHeaders(Hpack.java:511)
	at okhttp3.internal.http2.Http2Writer.headers(Http2Writer.java:299)
	at okhttp3.internal.http2.Http2Writer.synStream(Http2Writer.java:127)
	at okhttp3.internal.http2.Http2Connection.newStream(Http2Connection.java:230)
	at okhttp3.internal.http2.Http2Connection.newStream(Http2Connection.java:205)
	at okhttp3.internal.http2.Http2Codec.writeRequestHeaders(Http2Codec.java:111)
	at okhttp3.internal.http.CallServerInterceptor.intercept(CallServerInterceptor.java:50)
	at okhttp3.internal.http.RealInterceptorChain.proceed(RealInterceptorChain.java:147)
	at okhttp3.internal.connection.ConnectInterceptor.intercept(ConnectInterceptor.java:45)
	at okhttp3.internal.http.RealInterceptorChain.proceed(RealInterceptorChain.java:147)
	at okhttp3.internal.http.RealInterceptorChain.proceed(RealInterceptorChain.java:121)
	at okhttp3.internal.cache.CacheInterceptor.intercept(CacheInterceptor.java:93)
	at okhttp3.internal.http.RealInterceptorChain.proceed(RealInterceptorChain.java:147)
	at okhttp3.internal.http.RealInterceptorChain.proceed(RealInterceptorChain.java:121)
	at okhttp3.internal.http.BridgeInterceptor.intercept(BridgeInterceptor.java:93)
	at okhttp3.internal.http.RealInterceptorChain.proceed(RealInterceptorChain.java:147)
	at okhttp3.internal.http.RetryAndFollowUpInterceptor.intercept(RetryAndFollowUpInterceptor.java:125)
	at okhttp3.internal.http.RealInterceptorChain.proceed(RealInterceptorChain.java:147)
	at okhttp3.internal.http.RealInterceptorChain.proceed(RealInterceptorChain.java:121)
	at okhttp3.logging.HttpLoggingInterceptor.intercept(HttpLoggingInterceptor.java:212)
	at okhttp3.internal.http.RealInterceptorChain.proceed(RealInterceptorChain.java:147)
	at okhttp3.internal.http.RealInterceptorChain.proceed(RealInterceptorChain.java:121)
	at org.javacord.core.DiscordApiImpl.lambda$new$1(DiscordApiImpl.java:576)
	at org.javacord.core.DiscordApiImpl$$Lambda$25/000000000000000000.intercept(Unknown Source)
	at okhttp3.internal.http.RealInterceptorChain.proceed(RealInterceptorChain.java:147)
	at okhttp3.internal.http.RealInterceptorChain.proceed(RealInterceptorChain.java:121)
	at okhttp3.RealCall.getResponseWithInterceptorChain(RealCall.java:200)
	at okhttp3.RealCall.execute(RealCall.java:77)
	at org.javacord.core.util.rest.RestRequest.executeBlocking(RestRequest.java:355)
	at org.javacord.core.util.ratelimit.RatelimitManager.lambda$queueRequest$2(RatelimitManager.java:114)
	at org.javacord.core.util.ratelimit.RatelimitManager$$Lambda$42/000000000000000000.run(Unknown Source)
	at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:511)
	at java.util.concurrent.FutureTask.run(FutureTask.java:266)
	... 3 more

```

**Comment by Vampire** [comment]
Your class path is bad.
As you can see from your error message, you have `okio-1.6.0` in your classs path.
Javacord depends on `okhttp-3.9.1` which depends on `okio-1.13.0`
and on `xsalsa20poly1305-0.10.1` which depends on `okio-1.14.0`.

One of these is using a method in okio that is not yet available in 1.6.0, hence you get the `NoSuchMethodError` as the method was present during compiling but is missing now at runtime by you having 1.6.0 in the class path (first).

**Comment by Bastian** [comment]
As there was no further answer in the last 7 days, I will close this issue.
Feel free to create a new issue (or ask me to reopen this one) if you believe that this is a Javacord bug.

You can also join our Discord server if you need help :) (link is in README).
